File: app/src/main/java/com/simplemobiletools/draw/actions/Line.java
package com.simplemobiletools.draw.actions;

import android.graphics.Path;

import java.io.IOException;
import java.io.Writer;
import java.security.InvalidParameterException;

public final class Line implements Action {

    public final float x, y;

    public Line(String data) {
        if (data.startsWith("L"))
            throw new InvalidParameterException("The Line data should start with 'L'.");

        try {
            String[] xy = data.substring(1).split(",");
            x = Float.parseFloat(xy[0].trim());
            y = Float.parseFloat(xy[1].trim());
        } catch (Exception ignored) {
            throw new InvalidParameterException("Error parsing the given Line data.");
        }
    }

    public Line(float x, float y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public void perform(Path path) {
        path.lineTo(x, y);
    }

    @Override
    public void perform(Writer writer) throws IOException {
        writer.write("L");
        writer.write(String.valueOf(x));
        writer.write(",");
        writer.write(String.valueOf(y));
    }
}

File: app/src/main/kotlin/com/simplemobiletools/draw/actions/Line.kt
package com.simplemobiletools.draw.actions

import android.graphics.Path

import java.io.IOException
import java.io.Writer
import java.security.InvalidParameterException

class Line : Action {

    val x: Float
    val y: Float

    @Throws(InvalidParameterException::class)
    constructor(data: String) {
        if (data.startsWith("L"))
            throw InvalidParameterException("The Line data should start with 'L'.")

        try {
            val xy = data.substring(1).split(",".toRegex()).dropLastWhile(String::isEmpty).toTypedArray()
            x = java.lang.Float.parseFloat(xy[0].trim { it <= ' ' })
            y = java.lang.Float.parseFloat(xy[1].trim { it <= ' ' })
        } catch (ignored: Exception) {
            throw InvalidParameterException("Error parsing the given Line data.")
        }

    }

    constructor(x: Float, y: Float) {
        this.x = x
        this.y = y
    }

    override fun perform(path: Path) {
        path.lineTo(x, y)
    }

    @Throws(IOException::class)
    override fun perform(writer: Writer) {
        writer.write("L")
        writer.write(x.toString())
        writer.write(",")
        writer.write(y.toString())
    }
}
